File: databinding/src/main/java/com/example/anvil/databinding/SeekBarLayout.java
package com.example.anvil.databinding;

import android.content.Context;
import android.widget.LinearLayout;
import android.widget.SeekBar;

import static trikita.anvil.DSL.*;
import trikita.anvil.RenderableView;

public class SeekBarLayout extends RenderableView {

    int value = 0;

    public SeekBarLayout(Context c) {
        super(c);
    }

    @Override
    public void view() {
        linearLayout(() -> {
            size(FILL, FILL);
            orientation(LinearLayout.VERTICAL);

            seekBarLayout();
            seekBarLayout();
        });
    }

    private void seekBarLayout() {
        seekBar(() -> {
            size(FILL, WRAP);
            margin(dip(40));
            progress(value);
            onSeekBarChange((sb, progress, fromUser) -> {
                if (fromUser) {
                    value = progress;
                }
            });
        });
    }
}

File: databinding-kotlin/src/main/kotlin/com/example/anvil/databinding/SeekBarLayout.kt
package com.example.anvil.databinding

import android.content.Context
import android.widget.LinearLayout
import android.widget.SeekBar

import trikita.anvil.DSL.*
import trikita.anvil.RenderableView

class SeekBarLayout(c: Context) : RenderableView(c) {

    var value = 0

    override fun view() {
        linearLayout {
            size(FILL, FILL)
            orientation(LinearLayout.VERTICAL)

            seekBarLayout()
            seekBarLayout()
        }
    }

    private fun seekBarLayout() {
        seekBar {
            size(FILL, WRAP)
            margin(dip(40))

            progress(value)
            onSeekBarChange({ sb, progress, fromUser ->
                if (fromUser) {
                    value = progress.toInt()
                }
            })
        }
    }
}


